src/System.Private.ServiceModel/src/Internals/System/Runtime/GeneratedXmlSerializers.cs

```diff
@@ -12,18 +13,44 @@ namespace System.Runtime.Serialization
 {
     public class GeneratedXmlSerializers
     {
-        private static Dictionary<string, Type> generatedSerializers = new Dictionary<string, Type>();
+        private static Func<Dictionary<string, Type>> s_GeneratedSerializersInitializer;
+        private static Lazy<Dictionary<string, Type>> generatedSerializers = new Lazy<Dictionary<string, Type>>(InitGeneratedSerializers);
 
-        public static Dictionary<string, Type> GetGeneratedSerializers()
+        public static Func<Dictionary<string, Type>> GeneratedSerializersInitializer
         {
-            return generatedSerializers;
+            get
+            {
+                return s_GeneratedSerializersInitializer;
+            }
+            set
+            {
+                Contract.Assert(s_GeneratedSerializersInitializer == null, "s_GeneratedSerializersInitializer is already initialized.");
+                s_GeneratedSerializersInitializer = value;
+            }
+        }
+
+        private static Dictionary<string, Type> InitGeneratedSerializers()
+        {
+            if (GeneratedSerializersInitializer != null)
```

Comment:
It would be useful to include comments about what is happening here.  The combination of NET Native, UWP and CoreCLR code paths make the logic here not obvious to other developers.